apptools/feedback/feedbackbot/model.py

```diff
@@ -0,0 +1,100 @@
+"""
+This module implements a class that provides logic for a simple plugin
+for sending messages to a developer team's slack channel.
+"""
+
+import io
+import logging
+
+import slack
+from PIL import Image
+
+from traits.api import (
+    HasRequiredTraits, Str, Property, Array, String, Any)
+
+logger = logging.getLogger(__name__)
+
+
+class FeedbackMessage(HasRequiredTraits):
+    """ Model for the feedback message.
+
+    Notes
+    -----
+    The user-developer must specify the slack channel that the message must be
+    sent to, as well as provide raw screenshot data.
+
+    """
+
+    #: Name of the client user
+    name = Str(msg_meta=True)
+
+    #: Name of the client organization.
+    organization = Str(msg_meta=True)
+
+    # TODO: Slack supports some markdown in messages, provide
+    # some details here.
+    #: Main body of the feedback message.
+    description = Str(msg_meta=True)
+
+    #: The target slack channel that the bot will post to, must start with #
+    # and must be provided by the user-developer.
+    channels = String(minlen=2, regex='#.*', required=True)
+
+    #: OAuth token for the slackbot, must be provided by the user-developer.
+    token = Str(required=True)
+
+    #: The final message that gets posted to Slack.
+    msg = Property(Str, depends_on='msg_meta')
+
+    #: 3D numpy array to hold three channel (RGB) screenshot pixel data.
+    img_data = Array(shape=(None, None, 3), dtype='uint8', required=True)
+
+    # FIXME: Not sure if this the right way to go about initiating a
```

Comment:
Upon closer inspection, this is only used by the `send` method.  It's probably best to not make it a trait and create the buffer as needed inside of `send`.

Reply:
I moved it out assuming that it would reduce time spent in the `send` function. Some more experiments revealed that while time spent in `send` is reduced (1.57s to 1.5s), the difference is barely noticeable by the user, so now it's back in. (Note to self: time everything)